templates/settings-admin.php

```diff
@@ -18,62 +18,49 @@
  */
 
 /** @var \OCA\OAuth2\Db\Client $client */
-script('oauth2', 'settings');
+script('oauth2', 'settings-admin');
 style('oauth2', 'main');
 style('oauth2', 'settings-admin');
+
+$hasClients = !empty($_['clients']);
 ?>
 
 <div class="section" id="oauth2">
 	<h2 class="app-name"><?php p($l->t('OAuth 2.0')); ?></h2>
 
-    <h3><?php p($l->t('Registered clients')); ?></h3>
-    <?php if (empty($_['clients'])) {
-	p($l->t('No clients registered.'));
-} else {
-	?>
-    <table class="grid">
-        <thead>
-        <tr>
-            <th id="headerName" scope="col"><?php p($l->t('Name')); ?></th>
-            <th id="headerRedirectUri" scope="col"><?php p($l->t('Redirection URI')); ?></th>
-            <th id="headerClientIdentifier" scope="col"><?php p($l->t('Client Identifier')); ?></th>
-            <th id="headerSecret" scope="col"><?php p($l->t('Secret')); ?></th>
-			<th id="headerSubdomains" scope="col"><?php p($l->t('Subdomains allowed')); ?></th>
-            <th id="headerRemove">&nbsp;</th>
-        </tr>
-        </thead>
-        <tbody>
-            <?php foreach ($_['clients'] as $client) {
-		?>
-                <tr>
-                    <td><?php p($client->getName()); ?></td>
-                    <td><?php p($client->getRedirectUri()); ?></td>
-                    <td><code><?php p($client->getIdentifier()); ?></code></td>
-                    <td><code><?php p($client->getSecret()); ?></code></td>
-					<td id="td-allow-subdomains"><?php if ($client->getAllowSubdomains()) {
-			?> <img alt="" src="/core/img/actions/checkmark.svg"> <?php
-		} ?></td>
-                    <td>
-                        <form id="form-inline" class="delete" data-confirm="<?php p($l->t('Are you sure you want to delete this item?')); ?>" action="<?php p($_['urlGenerator']->linkToRoute('oauth2.settings.deleteClient', ['id' => $client->getId()])); ?>" method="post">
-							<input type="hidden" name="requesttoken" value="<?php p($_['requesttoken']) ?>" />
-                            <input type="submit" class="button icon-delete" value="">
-                        </form>
-                    </td>
-                </tr>
-            <?php
-	} ?>
-        </tbody>
-    </table>
-    <?php
-} ?>
+	<h3><?php p($l->t('Registered clients')); ?></h3>
+	<p class="no-clients-message <?php if ($hasClients) {
```

Comment:
any ideas for better readability?

Reply:
added `$noClientsExtraClass` and init it in the top of the templates.